### ISSUE TITLE
Provide images that support external remote rendering?

### ISSUE THREAD

**Issue by itziakos** [simphony/simphony-remote-docker#13]

**Comment by stefanoborini** [comment]
I explored virtualgl a bit, and I am not sure it does what we need.

VirtualGL assumes the application is running on the application server where the 3D acceleration is, and let the thin client receive only 2d requests. In our setup, our "thin client" is the docker image, which has no 3d acceleration. We would therefore need hardware with the 3d accelerator to which our docker  machine connects (e.g. via ssh) and starts the X11/3D application so that it gets rendered in the docker machine desktop. The application, however, is running on the remote 3d accelerated server, which means that we would have to provide applications there, to provide access to the users, and their filesystem would be the one of the remote machine, not of the docker container.
Basically, this is depicted in this image

![image](https://cloud.githubusercontent.com/assets/132967/20053693/a6f12a16-a4d1-11e6-9a1b-65ffd1d3b510.png)

I think that the assumption was that you could delegate 3d calls to a remote 3d rendering server. This is not performed by VirtualGL. We need Chromium for that (http://chromium.sourceforge.net/doc/LLNLcopy.html)

**Comment by stefanoborini** [comment]
I checked alternatives to chromium. X11 Cluster rendering is not a viable option

http://idav.ucdavis.edu/~okreylos/ResDev/Vrui/X11ClusterRendering.html

because it behaves like MPI: you start the application on multiple nodes, then it connects to your (e.g. docker) X11 and dispatches the rendering. This would require providing access to the rendering farm to the user, and the rendering farm must connect back to the docker container. So it's not an option.

**Comment by stefanoborini** [comment]
The problem with chromium is that it's basically unmaintained since a very long time, and its name makes it really hard to find relevant documentation. It is also unclear if it supports all advanced openGL features.

**Comment by stefanoborini** [comment]
Attempts to make chromium work failed. It installs properly, apart from basic missing dev packages.
To run it, the crdemo.conf has been used to start the mothership (central coordinator), where changes have been done to the server setup so that it referred to an external, 3d accelerated machine (Dell).

The crserver has then been started on the dell machine. It correctly connects to the mothership program, and negotiation is successful. 

Finally, invoking crappfake should, in principle, start the GL application with the rendering performed by the crserver. Unfortunately, this is not the case. the rendering keeps being performed locally.

It is unclear if this is due to an error somewhere in the setup, or in the fact that the project is unmaintained since 10 years and something might have changed so that it does not work anymore in modern environments. In any case, it's a risky strategy to rely on an abandoned project with no contact reference or mailing list.

I might give it another try, but I am unsure if it's a safe strategy to pursue anyway.

**Comment by itziakos** [comment]
@stefanoborini I totally agree that it is very risky to depend on unmaintained projects
